frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-details.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+
+import { getResource } from 'kubevirt-web-ui-components';
+
+import {
+  Firehose,
+  StatusBox,
+  ScrollToTopOnMount,
+  SectionHeading,
+} from '@console/internal/components/utils';
+
+import { TemplateKind } from '@console/internal/module/k8s';
+import { TemplateModel } from '@console/internal/models';
+import { VMTemplateResourceSummary, VMTemplateDetailsList } from './vm-template-resource';
+
+export const VMTemplateDetailsFirehose = ({ obj: template }: { obj: TemplateKind }) => {
+  const { name, namespace } = template.metadata;
+
+  const vmtRes = getResource(TemplateModel, {
+    name,
+    namespace,
+    isList: false,
+    prop: 'vmt',
+    optional: true,
+  });
+
+  const resources = [vmtRes];
+
+  return (
+    <div className="co-m-pane__body">
+      <Firehose resources={resources}>
+        <VMTemplateDetails template={template} />
+      </Firehose>
+    </div>
+  );
+};
+
+const VMTemplateDetails = (props: VMTemplateDetailsProps) => {
```

Comment:
please use the 
```tsx
VMTemplateDetails: React.FC<VMTemplateDetailsProps> 
``` 
convention

Reply:
done, @suomiy do we use this convention as default ?

Reply:
Not sure, I never saw a discussion about that. I prefer this one because it gives more specific type.

Reply:
In general, Console code should favor functional components over class components. The reason behind that is functions are (usually) simpler and easier to compose, i.e. favor composition over extension.

So with functional components, `React.FC<FooProps>` is the way to go.